### ISSUE TITLE
Cleanup batch_norm

### ISSUE THREAD

**Issue by nsorabaTT** [tenstorrent/tt-metal#19938]
1. Remove unused `stats` output [here](https://github.com/tenstorrent/tt-metal/blob/a5cf32bdc418b0122947c3991f0200df264cf2bc/ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp#L38)
2. Align batchnorm with torch equivalents: batchnorm2d, batchnorm3d etc.
3. Look into extending support for ranks other than 4.

**Comment by bbradelTT** [comment]
1. Was done recently.
2. Unless 2 and 3 are asked for we can leave things as is.

Closing.
